### DISCUSSION TITLE
A question to generate the face based model with IfcArbitraryOpenProfileDef

### DISCUSSION THREAD

**Discussion by bimintelligence** [IfcOpenShell/IfcOpenShell#2734] [Q&A]
Hi, I'm generating a surface like bellow using IFC.

So, I've select these entities (IfcArbitraryOpenProfileDef, IfcSurfaceOfLinearExtrusion),
which the ellipse entity is swept alongside the line.
However, the IFC file (attached below) is invalidated to be parsed.

![image](https://user-images.githubusercontent.com/73410751/216535772-d4fab2eb-a6a6-40f8-83ac-1a5ff8daedc8.png)

[IfcHouse.ifc.txt](https://github.com/IfcOpenShell/IfcOpenShell/files/10576430/IfcHouse.ifc.txt)

**Comment by aothms** [comment]
I don't see anything invalid, but as I indicated earlier advanced brep in IFC software may not be very mature.

Swept surfaces as part of adv brep can be tricky because it can introduce discontinuities in extruding a polyline for example (In ISO 10303-42 swept_surface is much purer because it's just a sweep of curve there not the profiledef in IFC).

But swept surfaces are explicitly allowed as part of IfcAdvancedFace so I guess implementation may need to adapt.

Did you try the same using a bspline surfaces instead?

**Comment by bimintelligence** [comment]
I've tried two methods to generate this face.
However, both of them failed.
Is there any solution to solve it.
![image](https://user-images.githubusercontent.com/73410751/216907318-5323c6f0-cb4a-4f1e-b7cf-7cb4ba24247c.png)
![image](https://user-images.githubusercontent.com/73410751/216907477-6d265926-8a67-40c4-ba89-57535f47fee4.png)

**Reply by aothms** [reply]
Question is again whether this is a model or a implementation issue. IfcOpenShell isn't great with adv brep, but a non-periodic explicit patch as sketched below should work. What output do you get in IfcConvert?
